colossalai/auto_parallel/solver/sharding_strategy.py

```diff
@@ -122,9 +126,9 @@ class ShardingStrategy_V2:
     """
     name: str
     sharding_specs: Dict[OperationData, ShardingSpec] = None
-    compute_cost: TrainCycleItem = None
-    communication_cost: TrainCycleItem = None
-    memory_cost: TrainCycleItem = None
+    compute_cost: TrainCycleItem[int] = None
```

Comment:
I am not sure about whether TrainCyclyItem object is subscriptable or not

Reply:
Fixed.